scripts/cli/util/userBalancePOLYX.js

```diff
@@ -0,0 +1,92 @@
+// Set options as a parameter, environment variable, or rc file.
+require = require("esm")(module /*, options*/);
+module.exports = require("./init.js");
+
+const { ExportToCsv } = require("export-to-csv");
+const cliProgress = require("cli-progress");
+let { reqImports } = require("./init.js");
+
+async function main() {
+	const api = await reqImports.createApi();
+	await getUserBalances(api);
+	process.exit();
+}
+
+// Gets user balance and stores them in an array
+async function getUserBalances(api) {
+	// create new progress bar
+	const progressBar = new cliProgress.SingleBar({
+		format: "progress [{bar}] {percentage}% || {value}/{total} || ETA: {eta_formatted}",
+		barCompleteChar: "\u2588",
+		barIncompleteChar: "\u2591",
+		hideCursor: true,
+	});
+	const listOfDids = await api.query.identity.didRecords.entries();
+	let empty_did = "0x0000000000000000000000000000000000000000000000000000000000000000";
+	let data = [];
+
+	// initializes the progress bar
+	progressBar.start(listOfDids.length - 1, 0, {
+		speed: "N/A",
+	});
+
+	for (let i = 0; i < listOfDids.length; i++) {
+		let pk = listOfDids[i][1]["primary_key"];
+		let listOfSecondaryKeyObjects = listOfDids[i][1]["secondary_keys"];
+
+		for (let j = 0; j < listOfSecondaryKeyObjects.length; j++) {
+			// secondary key
+			let sk = JSON.parse(listOfSecondaryKeyObjects[j]["signer"]).Account;
+			// get sk did
+			let skDid = await api.query.identity.keyToIdentityIds(sk);
+			// getting secondary key balance info
+			let skAccountData = await api.query.system.account(sk);
+			data.push({
+				AccountId: sk,
+				Identity: skDid,
+				free: JSON.stringify(skAccountData.data.free),
+				reserved: JSON.stringify(skAccountData.data.reserved),
+				miscFrozen: JSON.stringify(skAccountData.data.miscFrozen),
+			});
+		}
+
+		let did = await api.query.identity.keyToIdentityIds(pk);
+		let accountData = await api.query.system.account(pk);
+
+		if (did.toString() != empty_did) {
+			data.push({
+				AccountId: pk,
```

Comment:
I'd suggest adding an additional field:
`isPrimaryKey`
which is true / false depending on whether the AccountId is the primary key or a secondary key for the DID.

Reply:
I added the new field in my latest commit.